lib/auth/v4/queryAuthCheck.js

```diff
@@ -35,6 +37,11 @@ queryAuthCheck.check = (request, log, callback) => {
     const expiry = authParams.expiry;
     const credential = authParams.credential;
 
+    if (!areSignedHeadersComplete(signedHeaders, request.headers)) {
+        log.debug('signedHeaders are incomplete', { signedHeaders });
+        return { err: errors.AccessDenied };
```

Comment:
I find it very confusing to return something instead of using the callback like the other error cases. Is it intended? You may want to document that behaviour or provide only one type of exit.